presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/PinotColumnUtils.java

```diff
@@ -39,13 +51,57 @@ private PinotColumnUtils()
     {
         return pinotTableSchema.getColumnNames().stream()
                 .filter(columnName -> !columnName.startsWith("$")) // Hidden columns starts with "$", ignore them as we can't use them in PQL
-                .map(columnName -> new PinotColumn(columnName, getPrestoTypeFromPinotType(pinotTableSchema.getFieldSpecFor(columnName))))
+                .map(columnName -> new PinotColumn(columnName, getPrestoTypeFromPinotType(pinotTableSchema.getFieldSpecFor(columnName)), isNullableColumnFromPinotType(pinotTableSchema.getFieldSpecFor(columnName)),
+                        getCommentFromPinotType(pinotTableSchema.getFieldSpecFor(columnName))))
                 .collect(toImmutableList());
     }
 
+    private static String getCommentFromPinotType(FieldSpec field)
+    {
+        return field.getFieldType().name();
+    }
+
+    private static boolean isNullableColumnFromPinotType(FieldSpec field)
+    {
+        return DEFAULT_PINOT_COLUMN_IS_NULLABLE;
+    }
+
     public static Type getPrestoTypeFromPinotType(FieldSpec field)
     {
         if (field.isSingleValueField()) {
+            switch (field.getFieldType()) {
+                case TIME:
```

Comment:
I am wondering if there should be a config flag to guard this behavior ? It _may_ break Uber's use case. I hate adding extra flags but I think it would help with the rollout: the schema perceived by a user would now change post a rollout of this diff.

Reply:
Added two configs:
'pinot.infer-date-type-in-schema'
'pinot.infer-timestamp-type-in-schema'
 to switch on/off this type conversion.